students/pkoleyni/Lesson05/mailroom_3.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+import datetime
+import sys
+
+donor_list_dic = {'pooria koleyni': {'donation': [130]},
+                  'john adams': {'donation': [200, 340, 560]},
+                  'steve knot': {'donation': [50, 130]},
+                  'meru cheng': {'donation': [150, 430]},
+                  'maysam razm': {'donation': [25, 25]}
+                  }
+
+
+def check_add_name(name):
+    """
+    :param name: name of a donor to be added to the donor_list_dic of       donors if not already there
+                This function first verifies to see if name is not already in the donor_list_dic
+                Then it will add name to the donor_list_dic dictionary
+
+    """
+
+    if name not in donor_list_dic:
+        print('\"{}\" is not in the list of the donors, Adding to the list of donors'.format(name))
+        donor_list_dic[name] = {'donation': []}
+    else:
+        print('{} is already in the list of donors'.format(name))
+
+
+def donation_list(name):
+    """
+    :param name: name of a donor
+    :return: list of all the donation made by a donor
+    """
+    return donor_list_dic[name]['donation']
+
+
+def add_donation(name, donation):
+    """
+    :param name: name of a donor
+    :param donation: amount of donation made
+    :return:
+    """
+    donation_list(name).append(donation)
+    update_total_donation_to_dic(name)
+
+
+def update_total_donation_to_dic(name):
+    total_given = sum(donation_list(name))
+    donor_list_dic[name]['total_donation'] = total_given
+
+
+def thank_you():
+    """
+    This function is taking care of the required actions when user select thank you from the menu
+    :return:
+    """
+    usr_name_input = input('Enter full name of a donor or enter \"List\" for complete list of doors > ')
+    while usr_name_input == 'List':
+        [print(donor) for donor in donor_list_dic]
```

Comment:
this is not a correct pattern for comprehensions, you will have a list filled with None values since print() returns nothing.

here is what you can do:
`print("\n".join(list(donor_list_dic)))`